edx_rest_framework_extensions/auth/jwt/decoder.py

```diff
@@ -315,21 +327,21 @@ def _decode_and_verify_token(token, jwt_issuer):
     return decoded_token
 
 
-def _get_signing_jwk_key_set(jwt_issuer, add_symmetric_keys=True):
+def get_signing_jwk_key_set(secret_key, add_symmetric_keys=True, add_asymmetric_keys=True):
```

Comment:
[discuss] It feels odd to specify the secret key in the args, but not the asymmetric keys. It also feels odd to specify both the secret key and whether to use it at all.

What if the signature were instead `(asymmetric_keys, secret_key)`? The caller would just get the appropriate settings. If a value is None, it is not used. (I also would like to change the name before making it part of the public API—these are not signing keys, but verification keys.)

Reply:
Updated.